solver/src/settlement.rs

```diff
@@ -59,22 +60,105 @@ impl Trade {
     }
 }
 
-pub trait Interaction: std::fmt::Debug + Send {
+#[cfg_attr(test, mockall::automock)]
+pub trait Interaction: Debug + Send {
     // TODO: not sure if this should return a result.
-    // Write::write returns a result but we know we write to a vector in memory so we know it will
-    // never fail. Then the question becomes whether interactions should be allowed to fail encoding
-    // for other reasons.
     fn encode(&self) -> Vec<EncodedInteraction>;
 }
 
+#[cfg(test)]
+impl Debug for MockInteraction {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        f.write_str("MockInteraction")
+    }
+}
+
+pub trait UnwrapInteraction: Interaction {
+    fn token_address(&self) -> H160;
+    fn amount(&self) -> U256;
+
+    fn increase_amount(&mut self, amount: U256);
+}
+
+#[cfg(test)]
+mockall::mock! {
+    // Will create a `MockUnwrapInteraction` type.
+    UnwrapInteraction {}
+
+    impl Interaction for UnwrapInteraction {
+        fn encode(&self) -> Vec<EncodedInteraction>;
+    }
+    impl UnwrapInteraction for UnwrapInteraction {
+        fn token_address(&self) -> H160;
+        fn amount(&self) -> U256;
+        fn increase_amount(&mut self, amount: U256);
+    }
+}
+
+#[cfg(test)]
+impl Debug for MockUnwrapInteraction {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        f.write_str("MockUnwrapInteraction")
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct SettlementInteractions {
+    amm_interactions: Vec<Box<dyn Interaction>>,
+    unwrap_interactions: HashMap<H160, Box<dyn UnwrapInteraction>>,
```

Comment:
I find these name quite specific (it's true now but rather coincidental to the reason for these fields). Maybe call them `execution_plan`, `pre_transfer_interactions` or something like that?